src/core/row.cpp

```diff
@@ -109,3 +137,156 @@ std::size_t row::find_column(std::string const &name) const
 
     return it->second;
 }
+
+std::pair<std::string*, indicator*> row::alloc_data_holder_string()
+{
+    data_holder *holder = NULL;
+    std::string *s = new std::string();
+    try
+    {
+        holder = new data_holder;
+        holder->type = dt_string;
+        holder->data.p = s;
+        holder->ind = i_ok;
+        holders_.push_back(holder);
+        return std::make_pair(s, &holders_.back()->ind);
+    } catch (...) {
+        delete holder;
+        delete s;
+        throw;
+    }
```

Comment:
And this (and all the other similar blocks below) should be removed once the pointers above are replaced with `auto_ptr`.

OTOH, to be honest, I don't really see what exceptions could possibly happen here anyhow (`bad_alloc`? this one is not really recoverable from in any case...), so I'm not even sure why was this done in the first place?